src/Proto3/Wire/Decode.hs

```diff
@@ -393,16 +385,20 @@ byteString = bytes
 lazyByteString :: Parser RawPrimitive BL.ByteString
 lazyByteString = fmap BL.fromStrict bytes
 
--- | Parse a primitive with the @bytes@ wire type as 'Text'.
+-- | Decode a length-delimited wire field into a 'String'.
+string :: Parser RawPrimitive String
+string = Parser $ \case
+  LengthDelimitedField bs -> pure (ByteString.UTF8.toString bs)
+  other -> throwWireTypeError "String" other
+
+-- | Parse a primitive with the "bytes" wire type as 'Text'.
 text :: Parser RawPrimitive Text
-text = Parser $
-    \case
-        LengthDelimitedField bs ->
-            case decodeUtf8' $ BL.fromStrict bs of
-                Left err -> Left (BinaryError (pack ("Failed to decode UTF-8: " ++
-                                                         show err)))
-                Right txt -> return txt
-        wrong -> throwWireTypeError "string" wrong
+text = Parser $ \case
+  LengthDelimitedField bs ->
+    case decodeUtf8' $ BL.fromStrict bs of
+      Left err -> Left (BinaryError (pack ("Failed to decode UTF-8: " ++ show err)))
+      Right txt -> return txt
+  wrong -> throwWireTypeError "Text" wrong
```

Comment:
This change seems to be incorrect.  What is being discussed in the error message is wire format and perhaps protobuf field type ("string"), not how we represent types internally within a Haskell program ("Text").  In this case, we are distinguishing between type 2 and other wire types:

https://developers.google.com/protocol-buffers/docs/encoding#structure